### PR TITLE
fix(cohorts): Support precalculated-cohorts

### PR DESCRIPTION
**Pull request by neilkakkar**
## Problem

This PR:

1. Adds tests for precalculated cohorts
2. Adds tests for negation cohorts validation
3. Makes negation validation more precise, and makes it work with simplify filters
4. Introduces `exact` operator alongside `eq`

## Changes

👉 *Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review.*

### CODE REVIEW

File: ee/clickhouse/queries/cohort_query.py [hunk 1]
```diff
@@ -212,6 +212,9 @@ def _build_sources(self, subq: List[Tuple[str, str]]) -> Tuple[str, str]:
                     subq_alias == self.PERSON_TABLE_ALIAS
                     and "person" not in [prop.type for prop in getattr(self._outer_property_groups, "flat", [])]
                     and "cohort" not in [prop.type for prop in getattr(self._outer_property_groups, "flat", [])]
+                    and "static-cohort" not in [prop.type for prop in getattr(self._outer_property_groups, "flat", [])]
```
Comment (neilkakkar) [inline ee/clickhouse/queries/cohort_query.py:215]: need to fix this properly

File: frontend/src/scenes/cohorts/MatchCriteriaSelector.tsx [hunk 2]
```diff
@@ -88,10 +88,7 @@ function PropertyCriteriaRow({
                     }}
                     propertyFilters={group.properties}
                     style={{ margin: '1rem 0 0' }}
-                    taxonomicGroupTypes={[
-                        TaxonomicFilterGroupType.PersonProperties,
-                        TaxonomicFilterGroupType.CohortsWithAllUsers,
-                    ]}
+                    taxonomicGroupTypes={[TaxonomicFilterGroupType.PersonProperties, TaxonomicFilterGroupType.Cohorts]}
```
Comment (neilkakkar) [inline frontend/src/scenes/cohorts/MatchCriteriaSelector.tsx:91]: no more 'all users' in cohort selection!

File: ee/clickhouse/queries/cohort_query.py [hunk 3]
```diff
@@ -323,7 +326,7 @@ def _get_condition_for_property(self, prop: Property, prepend: str, idx: int) ->
                 res, params = self.get_performed_event_regularly(prop, prepend, idx)
         elif prop.type == "person":
             res, params = self.get_person_condition(prop, prepend, idx)
-        elif prop.type == "cohort":
+        elif prop.type in ["cohort", "static-cohort", "precalculated-cohort"]:
```
Comment (EDsCODE) [inline ee/clickhouse/queries/cohort_query.py:329]: https://github.com/PostHog/posthog/pull/9704 should handle "cohort" and "precalculated-cohort" by just unwrapping the properties so they're directly handled

Comment (EDsCODE) [inline ee/clickhouse/queries/cohort_query.py:329]: https://github.com/PostHog/posthog/pull/9704 should handle "cohort" and "precalculated-cohort" by just unwrapping the properties so they're directly handled

**Review by EDsCODE — APPROVED** [review]

**Pull request merged**